### PR TITLE
EPMRPP-77501 || sorted integrations after add new item

### PR DESCRIPTION
**Pull request by KrasovskyAlexander**
[EPMRPP-77501](https://jiraeu.epam.com/browse/EPMRPP-77501)
sorted integrations by creationDate after add new item

### CODE REVIEW

File: app/src/pages/inside/projectSettingsPageContainer/content/integrations/integrationsList/integrationInfo/integrationCollection/integrationCollection.jsx [hunk 1]
```diff
@@ -23,10 +23,12 @@ import styles from './integrationCollection.scss';
 const cx = classNames.bind(styles);
 
 export const IntegrationCollection = ({ items, disabled, openIntegration }) => {
+  const sortedItems = items.sort((a, b) => b.creationDate - a.creationDate);
```
Comment (Bam6ycha) [inline app/src/pages/inside/projectSettingsPageContainer/content/integrations/integrationsList/integrationInfo/integrationCollection/integrationCollection.jsx:26]: Do we really need to sort that array of integrations on every re-renders?

Could it be better to create an effect? Which runs on componentdidmount, and sort that array inside that effect. I suggest you use the useLayoutEffect hook here.

**Review by Bam6ycha — CHANGES_REQUESTED** [review]

**Review by AmsterGet — CHANGES_REQUESTED** [review]
@KrasovskyAlexander @Bam6ycha 
As I see in the previous task (`EPMRPP-67609`), the sorting was done on the BE side, so it will be redundant to sort integrations again on UI.
I would suggest to just add the new integration to the start of the list at `src/controllers/plugins/reducer.js` `addIntegration` function.

File: app/src/pages/inside/projectSettingsPageContainer/content/integrations/integrationsList/integrationInfo/integrationCollection/integrationCollection.jsx [hunk 2]
```diff
@@ -23,12 +23,10 @@ import styles from './integrationCollection.scss';
 const cx = classNames.bind(styles);
 
 export const IntegrationCollection = ({ items, disabled, openIntegration }) => {
-  const sortedItems = items.sort((a, b) => b.creationDate - a.creationDate);
-
   return (
```
Comment (Bam6ycha) [inline app/src/pages/inside/projectSettingsPageContainer/content/integrations/integrationsList/integrationInfo/integrationCollection/integrationCollection.jsx:26]: Optional:
I think here we can delete `return` statement.

Comment (AmsterGet) [reply app/src/pages/inside/projectSettingsPageContainer/content/integrations/integrationsList/integrationInfo/integrationCollection/integrationCollection.jsx]: Done.

**Review by Bam6ycha — APPROVED** [review]

**Pull request merged**